src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -167,7 +167,7 @@ case class AvroRelation(location: String)(@transient val sqlContext: SQLContext)
             null
           } else {
             // Avro map keys are always strings, so it's enough to just call toString on them.
-            item.asInstanceOf[Map[Any, Any]].map(x => (x._1.toString, valueConverter(x._2)))
+            item.asInstanceOf[Map[Any, Any]].map(x => (x._1.toString, valueConverter(x._2))).toMap
```

Comment:
This isn't quite related to this PR, but is sort of a bug fix. Apparently, before this change we would get a HashMap, and while it didn't cause any troubles for avro <-> sparkSQL, when I tried to read an avro file and save it as parquet, scala Map was expected. This fixes that. Let me know if I should make a separate PR for this though.